bin/setup_new

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+# taken from https://gist.github.com/gvarela/9370838
+
+{
+  RUBY_VERSION=$(cat .ruby-version)
+  RUBY_VERSION=${RUBY_VERSION:=2.3.0}
+  PROJECT_NAME="ifme"
+} >/dev/null 2>&1
+
+## bootstrap function called at bottom of file comment out verification steps you don't need
+init (){
+  verify_ruby_version &&
+    verify_bundled_gems
+  verify_env_file
+  copy_example_config_files
+  verify_postgres_running &&
+    verify_database_migrations
+}
+
+
+## utility functions to test dependencies
+is_darwin (){
+  PLATFORM=$(uname)
+  [ "$PLATFORM" == "Darwin" ]
+}
+
+has_homebrew (){
+  [ is_darwin ] && [ hash brew >/dev/null 2>&1 ]
+}
+
+verify_homebrew (){
+  if [ ! has_homebrew ]
+  then
+    if [ is_darwin ]
+    then
+      echo 'Please install homebrew. Install instructions at http://brew.sh/'
+    else
+      echo "Please install Ruby version $RUBY_VERSION"
+    fi
+    exit 1
+  fi
+}
+
+has_postgres (){
+  hash psql >/dev/null 2>&1
```

Comment:
I didn't know the answer to that either, so I did some research and according to [this stackoverflow](http://stackoverflow.com/questions/592620/check-if-a-program-exists-from-a-bash-script)  `hash` is the recommended way to check that a program exists when `bash` is your shell/hashbang (as it is in our case). It is similar to `which`, except it is better to use `hash` because:

* `which` is an external process, so it is cheaper to use hash because it is builtin
* Many operating systems have a `which` that doesn't even set an exit status
* Many operating systems make `which` do custom and evil stuff like change the output or even hook into the package manager.
* `hash` has the side-effect that the command's location will be hashed (for faster lookup next time you use it), which is usually a good thing since you probably check for its existence in order to actually use it.